userspace/libsinsp/sinsp_ppm_sc.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <sinsp.h>
+
+std::unordered_set<uint32_t> sinsp::simple_ppm_sc_set()
+{
+	return std::unordered_set<uint32_t>{
+		PPM_SC_ACCEPT,
+		PPM_SC_ACCEPT4,
+		PPM_SC_BIND,
+		PPM_SC_BPF,
+		PPM_SC_CAPSET,
+		PPM_SC_CHDIR,
+		PPM_SC_CHMOD,
+		PPM_SC_CHROOT,
+		PPM_SC_CLONE,
+		PPM_SC_CLONE3,
+		PPM_SC_CLOSE,
+		PPM_SC_CONNECT,
+		PPM_SC_COPY_FILE_RANGE, // REMOVE?
+		PPM_SC_CREAT,
+		PPM_SC_DUP,
+		PPM_SC_DUP2,
+		PPM_SC_DUP3,
+		PPM_SC_EVENTFD,
+		PPM_SC_EVENTFD2,
+		PPM_SC_EXECVE,
+		PPM_SC_EXECVEAT,
+		PPM_SC_FCHDIR,
+		PPM_SC_FCHMOD,
+		PPM_SC_FCHMODAT,
+		PPM_SC_FCNTL, // 64 TOO?
+		PPM_SC_FLOCK,
+		PPM_SC_FORK,
+		PPM_SC_GETSOCKOPT, // do we need this?
```

Comment:
This is a getter; do we need it?

Reply:
We're snooping the returned value to determine socket state (pending/established/failed IIRC). I don't know if Falco uses it (could be nice to detect firewalled off connection attempts, maybe?), the Sysdig agent does.

Reply:
Thanks @gnosek :heart:

Reply:
The feature is enabled at runtime (`sinsp_parser::set_track_connection_status(bool enabled)`) so it might be feasible to enable PPM_SC_GETSOCKOPT based on that?